### PR TITLE
DPL-683 - Added page origin + timestamp to traction alerts

### LINKED ISSUE #1082: DPL-683 Add page origin and time to traction alerts (BP-Low ¦ Could've)
**User story**
As a user (James W) I would alerts in Traction to appear with the page the alert came from and the time the alert was shown. This will help distinguish alerts across pages.

**Who are the primary contacts for this story**
Ben T

**Who is the nominated tester for UAT**
James W

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Traction alerts should have a header showing the origin page and time the alert was shown

Example from James:
image

### PR DESCRIPTION
**Pull request by SHIV5T3R**
Closes #1082 

#### Changes proposed in this pull request

- A footer with the page origin and timestamp for traction alerts

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: src/components/TractionMessage.vue [hunk 1]
```diff
@@ -62,6 +65,16 @@ export default {
       type: String,
       required: true,
     },
+    origin: {
+      type: String,
+      required: false,
+      default: 'Unknown Origin',
```
Comment (BenTopping) [inline src/components/TractionMessage.vue:71]: I think if the origin is unknown we should just show the time.

File: src/components/TractionMessage.vue [hunk 2]
```diff
@@ -62,6 +65,16 @@ export default {
...
+    },
+    time: {
+      type: String,
+      required: false,
+      default: new Date().toLocaleTimeString(),
```
Comment (BenTopping) [inline src/components/TractionMessage.vue:76]: I dont think we need the date of the alert, just the time like 'HH:MM:SS'

File: src/components/TractionMessage.vue [hunk 3]
```diff
@@ -10,19 +10,22 @@
...
-      {{ message }}
-    </div>
-    <div class="flex justify-end">
-      <button data-attribute="close" @click="dismiss">
+    <div class="flex justify-end items-center">
+      <button data-attribute="dismiss" @click="dismiss">
         <traction-close-icon :class-names="`${color.icon}`" />
       </button>
     </div>
+    <div class="inline-block flex-grow text-end" style="margin-bottom: 3px">
```
Comment (BenTopping) [inline src/components/TractionMessage.vue:23]: style="margin-bottom:3px" can be replaced by a tailwind class - `mb-2`

**Review by BenTopping — COMMENTED** [review]
Works well, nice one.

**Comment by SHIV5T3R** [comment]
@BenTopping ty for the review, made those changes ^^^^

**Review by seenanair — APPROVED** [review]
Looking good 👍

**Review by BenTopping — APPROVED** [review]

**Pull request merged**